### ISSUE TITLE
run-build-script error: TARGET

### ISSUE THREAD

**Issue by ahirner** [indygreg/PyOxidizer#366]
I'm trying to get a custom embedded python build to work.

## Setup
```bash
$ uname -rs
Darwin 19.6.0
$ pyoxidizer --version |head -n2
PyOxidizer 0.11.0-pre
commit: 7d00d53945167640a4f0db801aea97e492baad39
$ pyoxidizer init-rust-project test_build
     Created binary (application) package
<snip>
$ cd test_build/
```

## Build with build.rs errors
Here, I would customize `src/main.rs`. For now just invoke `run-build-script` on the vanilla project.
```bash
test_build/ $ pyoxidizer run-build-script build.rs
cargo:rerun-if-changed=build.rs
cargo:rerun-if-env-changed=PYOXIDIZER_CONFIG
error: TARGET
```

## Seeking possible causes
It seems the build context is not passed or created correctly. I try to substitute but am not sure if done properly.
```bash
test_build $ OUT_DIR=./build PROFILE=debug CARGO_MANIFEST_DIR=./ TARGET=exe PYOXIDIZER_CONFIG=pyoxidizer.bzl pyoxidizer run-build-script build.rs
cargo:rerun-if-changed=build.rs
cargo:rerun-if-env-changed=PYOXIDIZER_CONFIG
using PyOxidizer config file from PYOXIDIZER_CONFIG: pyoxidizer.bzl
cargo:rerun-if-changed=pyoxidizer.bzl
no existing PyOxidizer artifacts found
resolving 1 targets
resolving target resources
resolving target exe
resolving target dist
error[PYOXIDIZER_BUILD]: could not find default Python distribution for exe
 --> pyoxidizer.bzl:9:12
  |
9 |     return default_python_distribution()
  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ default_python_distribution()


error: could not find default Python distribution for exe
```

Maybe this issue is related to #355 which also reproduces on my setup.

**Comment by indygreg** [comment]
The issue here is that `TARGET=exe` is setting the target triple (where the binary will run on) to `exe` instead of a Rust target triple.

This is coming from the `let target = env::var("TARGET").context("TARGET")?;` line in `project_building.rs`.

I _think_ this is user error by specifying a _bad_ `TARGET` value. But it is possible I misunderstood what `TARGET` means or there are some docs specifying to use `TARGET` incorrectly. What brought you to type `TARGET=exe`?

**Comment by ahirner** [comment]
I'm not sure where I (mis)read about `TARGET`. I also can't make sense why manually setting all or some envs is required or what their exact value should be. Specifying the correct `TARGET` gets further.
```bash
test_build $ OUT_DIR=./build PROFILE=debug CARGO_MANIFEST_DIR=./ TARGET=x86_64-apple-darwin PYOXIDIZER_CONFIG=pyoxidizer.bzl pyoxidizer run-build-script build.rs
...
cargo:warning=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: /var/folders/lc/1ft4tsvn3j9884ct6tyrlbs00000gp/T/pyoxidizer-build-exe-packagingTH93SU/libpythonXY.a(dlfcn_simple.o) has no symbols
exit code: 0
libpythonXY created
error: error calling build(): Runtime(RuntimeError { code: "PYOXIDIZER", message: "writing packed resources", label: "build()" })
```
Is this still helpful?

**Comment by indygreg** [comment]
That `writing packed resources` error was fixed yesterday and was attributed to #390.

Are you still running into issues here or can this issue be closed?

**Comment by ahirner** [comment]
I was able to perform the last command successfully with `0.14.1`. It asked and needed an upgrade to macosx sdk 11 / `Darwin 20.4.0`. Thanks for the new release!
